posthog/tasks/update_cache.py

```diff
@@ -203,64 +274,91 @@ def update_cache_item(key: str, cache_type: CacheType, payload: dict) -> List[Di
+    if _cache_includes_latest_events(payload, filter):
+        cache.touch(key, timeout=settings.CACHED_RESULTS_TTL)
+        cache_update_reporting.on_results("update_cache_item_can_skip_because_events_do_not_invalidate_cache")
     else:
-        insights_queryset.update(last_refresh=timezone.now(), refreshing=False)
-        dashboard_tiles_queryset.update(last_refresh=timezone.now(), refreshing=False)
-        statsd.incr(
-            "update_cache_item_no_results",
-            tags={"team": team_id, "cache_key": key, "insight_id": insight_id, "dashboard_id": dashboard_id},
-        )
-        _mark_refresh_attempt_when_no_results(dashboard_id, dashboard_tiles_queryset, insight_id, insights_queryset)
-        result = []
+        try:
+            if (dashboard_id and dashboard_tiles_queryset.exists()) or insights_queryset.exists():
+                result = _update_cache_for_queryset(cache_type, filter, key, team)
+        except Exception as e:
+            cache_update_reporting.on_query_error(e)
+            raise e
+
+        if result:
+            cache_update_reporting.on_results("update_cache_item_success")
+        else:
+            cache_update_reporting.on_no_results()
+            result = []
 
     logger.info(
         "update_insight_cache.processed_item",
         insight_id=payload.get("insight_id", None),
         dashboard_id=payload.get("dashboard_id", None),
         cache_key=key,
-        has_results=len(result) > 0,
+        has_results=result and len(result) > 0,
     )
 
     return result
 
 
-def _mark_refresh_attempt_when_no_results(
-    dashboard_id: Optional[int],
-    dashboard_tiles_queryset: QuerySet,
-    insight_id: Union[int, str],
-    insights_queryset: QuerySet,
-) -> None:
-    if insights_queryset.exists() or dashboard_tiles_queryset.exists():
-        _mark_refresh_attempt_for(insights_queryset)
-        _mark_refresh_attempt_for(dashboard_tiles_queryset)
-    else:
-        if insight_id != "unknown":
-            _mark_refresh_attempt_for(
-                Insight.objects.filter(id=insight_id)
-                if not dashboard_id
-                else DashboardTile.objects.filter(insight_id=insight_id, dashboard_id=dashboard_id)
-            )
+def _cache_includes_latest_events(
+    payload: Dict, filter: Union[RetentionFilter, StickinessFilter, PathFilter, Filter]
+) -> bool:
+    """
+    event_definition has last_seen_at timestamp
+    a cacheable has last_refresh
+
+    if redis has cached result (is this always true with last_refresh?)
+    and last_refresh is after last_seen_at for each event in the filter
+
+    then there's no point re-calculating
+    """
+
+    # dashboards mostly use trends and funnels so concentrate on items that have filter.events to interrogate
+
+    last_refresh = payload.get("last_refresh", None)
+    if last_refresh:
+        event_ids = _events_from_filter(filter)
```

Comment:
`event_ids` made me think of `Event.id`, when this is `Event.event`, so maybe `event_names`?